bentoml/_internal/bento/build_config.py

```diff
@@ -225,6 +226,13 @@ def write_to_bento(
                 setup_script = resolve_user_filepath(self.setup_script, build_ctx)
             except FileNotFoundError as e:
                 raise InvalidArgument(f"Invalid setup_script file: {e}")
+            if not os.access(setup_script, os.X_OK):
```

Comment:
is this required?  don't we do `chmod +x` in the dockerfile with the changes below? @aarnphm @sauyon

Reply:
The `chmod` in the dockerfile are just docker related, we want to have a executable check here so that users make sure the script is runnable.

Reply:
@aarnphm it's not necessary, right? since we only run that script in docker?

Reply:
I think the intention for this is to check whether a script is executable or not. But that evolve using a library like [python-magic](https://github.com/ahupp/python-magic). I think we end up just doing this as a first pass.

Reply:
I'm not entirely sure what we want to do here. But the check is indeed not required.

Reply:
🤔